v0/destinations/webengage/transform.js

```diff
@@ -0,0 +1,124 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  WEBENGAGE_IDENTIFY_EXCLUSION,
+  BASE_URL,
+  BASE_URL_IND
+} = require("./config");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  defaultPostRequestConfig,
+  CustomError,
+  ErrorMessage,
+  extractCustomFields,
+  getValueFromMessage
+} = require("../../util");
+const set = require("set-value");
+const moment = require("moment");
+const { isValidTimestamp } = require("../ometria/util");
+
+const responseBuilder = (message, category, { Config }) => {
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  if (!payload.userId && !payload.anonymousId) {
+    throw new CustomError(
+      "[WEBENGAGE]: Either one of userId or anonymousId is mandatory.",
+      400
+    );
+  }
+  let baseUrl, endpoint;
+  if (Config.dataCenter === "ind") {
+    baseUrl = BASE_URL_IND;
+  } else {
+    baseUrl = BASE_URL;
+  }
+
+  if (category.type === "identify") {
+    const customAttributes = {};
+    extractCustomFields(
+      message,
+      customAttributes,
+      ["context.traits", "traits"],
+      WEBENGAGE_IDENTIFY_EXCLUSION
+    );
+
+    payload = { ...payload, attributes: customAttributes };
+    endpoint = `${baseUrl}/${Config.licenseCode}/users`;
+  } else {
+    let eventTimeStamp = message?.properties?.eventTime;
```

Comment:
qq -> why are we taking timestamp from properties? why not originaltimestamp?

Reply:
Yeah, changed now, we would take from timestamp or originalTimestamp.

Reply:
Should we define a priority order?
properties.eventTime > originalTimestamp > timestamp

Reply:
Yeah, we set some priority order, but now we mapped it to timestamp or originalTimestamp

Reply:
What's the benefit of having `properties.eventTime`?

Reply:
eventTime is the user field for taking the timestamp, now we take the timestamp from "timestamp" or "originalTimestamp"

Reply:
The latest changes are just using `eventTime`. What's the final stand on this?

Reply:
We are not using eventTime from properties. We have put the eventTime in WEBENGAGETrackConfig.json. There we are taking it from genericMappingField for timestamp.